tools/stress-cluster/services/Stress.Watcher/src/ChaosResources/GenericChaosClient.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using k8s;
+
+namespace Stress.Watcher
+{
+    public class GenericChaosClient
+    {
+        // Plural names for listing resource instances via Kubernetes API
+        public enum ChaosResourcePlurals
+        {
+            networkchaos,
+            stresschaos,
+            httpchaos,
+            iochaos,
+            kernelchaos,
+            timechaos,
+            jvmchaos
+        };
+
+        public string Group = "chaos-mesh.org";
+        public string Version = "v1alpha1";
+
+        private List<GenericClient> Clients = new List<GenericClient>();
+
+        public GenericChaosClient(KubernetesClientConfiguration config)
+        {
+            foreach (var plural in Enum.GetValues(typeof(ChaosResourcePlurals)))
+            {
+                Clients.Add(new GenericClient(config, Group, Version, plural.ToString()));
+            }
+        }
+
+        public async Task<CustomResourceList<GenericChaosResource>> ListNamespacedAsync(string ns)
+        {
+            var tasks = Clients.Select(async cl => await cl.ListNamespacedAsync<CustomResourceList<GenericChaosResource>>(ns));
```

Comment:
```suggestion
            var tasks = Clients.Select(cl => cl.ListNamespacedAsync<CustomResourceList<GenericChaosResource>>(ns));
```
I'm not sure but does that work without the async/await calls?

Reply:
It runs fine. My understanding is that this will return the `Task` value immediately, and then `await` them on the line below with `await Task.WhenAll(tasks);`?

Reply:
My expectation is that you would collect all the Task objects and then the await would happen as part of Task.WhenAll happens.